src/services/storage/hooks/offers.hooks.ts

```diff
@@ -22,7 +23,53 @@ export default {
         }
       }
     ],
-    find: [],
+    find: [
+      (context: HookContext) => {
+        if (context.params.query) {
+          const { averagePrice, totalCapacity, periods, provider, $limit } = context.params.query
+
+          if (!$limit) {
+            context.params.sequelize = {
+              raw: false,
+              nest: true,
+              include: [
+                {
+                  model: BillingPlan,
+                  as: 'plans'
+                },
+                {
+                  model: Agreement
+                }
+              ],
+              order: [['plans', 'period', 'ASC']],
+              where: {}
+            }
+
+            if (provider && provider.$like) {
+              context.params.sequelize.where.provider = {
+                [Op.like]: `%${provider.$like}%`
+              }
+            }
+
+            if (averagePrice) {
+              context.params.sequelize.where.averagePrice = {
+                [Op.between]: [averagePrice.min, averagePrice.max]
+              }
+            }
+
+            if (totalCapacity) {
+              context.params.sequelize.where.totalCapacity = {
+                [Op.between]: [totalCapacity.min, totalCapacity.max]
```

Comment:
Is this filter works???
`totalCapacity` is gype of `string`

Reply:
no no of course not, as as you pointed out the totalCapacity is a type string.

Reply:
Now it should work fine :) Thanks for finding the casting process @nduchak.

Reply:
Looks cool. Thank you 👍

Reply:
Hmmm.. 
But this `totalCapacity.min` and `totalCapacity.max` coming from request? It's a possible SQL injection
You probably should sanitize or add some validation that these fields are actually a number